digits/extensions/data/imageProcessing/forms.py

```diff
@@ -0,0 +1,96 @@
+# Copyright (c) 2016, NVIDIA CORPORATION.  All rights reserved.
+from __future__ import absolute_import
+
+import os
+
+from flask.ext.wtf import Form
+from wtforms import validators
+
+from digits import utils
+from digits.utils import subclass
+from digits.utils.forms import validate_required_iff
+
+
+@subclass
+class DatasetForm(Form):
+    """
+    A form used to create an image processing dataset
+    """
+
+    def validate_folder_path(form, field):
+        if not field.data:
+            pass
+        else:
+            # make sure the filesystem path exists
+            if not os.path.exists(field.data) or not os.path.isdir(field.data):
+                raise validators.ValidationError(
+                    'Folder does not exist or is not reachable')
+            else:
+                return True
+
+    source_folder = utils.forms.StringField(
+        u'Source image folder',
+        validators=[
+            validators.DataRequired(),
+            validate_folder_path,
+            ],
+        tooltip="Indicate a folder full of images."
+        )
+
+    target_folder = utils.forms.StringField(
+        u'Target image folder',
+        validators=[
+            validators.DataRequired(),
+            validate_folder_path,
+            ],
+        tooltip="Indicate a folder full of images. There must be one "
+                "image per image in the source image folder. Image "
+                "names do not matter but the images should match "
+                "those of the source folder, when sorted alphanumerically."
```

Comment:
I agree... I made it like this initially because in the first dataset I used all images were one off e.g. img1.jpg in target folder corresponded to img0.jpg in source folder. I can change this.